implementing-nosto/implement-search/getting-started-with-nosto-search/implementing-category-pages.md

```diff
@@ -49,6 +45,28 @@ init({
 ```
 {% endcode %}
 
+### Handling native results
+
+It's recommended to hide the original category page products to avoid displaying them while Nosto is initializing. This prevents any possible confusion for users and avoids visual flickering of the page.
+
+The best approach is to add `ns-content-hidden` class name to the same element you are targeting with `contentCssSelector` or `categoryCssSelector`. This class name will be stripped away by Nosto automatically as soon as the script is initialized.
+
+In addition, you should define CSS to hide the target element:
+
+{% code title="css" %}
+```css
+.ns-content-hidden {
+    display: none;
+}
+```
+{% endcode %}
+
+{% hint style="info" %}
+You are free to change the exact styles use as long as they hide the content from view and the class name matches the one Nosto is looking for.
+{% endhint %}
+
+It is not recommended to completely block or remove native results from the page as Nosto will attempt to display the original category page products in case Nosto service is unavailable or can't be reached. In addition, the original products are made available for the SEO crawlers, improving your page's ranking in the search engines.
```

Comment:
Consider,
instead of.
_improving your page's ranking in the search engines._
do this
improving the page's ranking in the search engines.

Reply:
Remove ""It's not recommended" line and move "Nosto will attempt" section to the top [here](https://github.com/Nosto/wiki-docs/pull/154/files#diff-974f3c57b1d58a2ba40ca0e222b90fdb5b554541eaa55b0386a00c3cc88ed1d8R49) and then explain 
_So, It's recommended to hide the original category page products, instead of removing or blocking them, etc.._